packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -1016,15 +1018,14 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
   ): Promise<{ ok: number }> {
     const auth: Auth = { username: authDoc.user, password: authDoc.pwd };
     // NOTE: we keep all the original options and just overwrite the auth ones.
-    const clientOptions: MongoClientOptions = {
-      ...DEFAULT_DRIVER_OPTIONS,
+    const clientOptions = processDriverOptions({
       ...this.initialOptions,
-      auth: auth
-    };
+      auth
+    });
     if (authDoc.mechanism) clientOptions.authMechanism = authDoc.mechanism as AuthMechanismId;
     if (authDoc.authDb) clientOptions.authSource = authDoc.authDb;
     const mc = await MongoClient.connect(
-      this.uri as string,
+      Object.assign(new URL(this.uri as string), { username: '', password: '' }).href,
```

Comment:
Also waiting for the driver team on slack here

Reply:
So, it looks like this was intentional on their end. Is `this.uri` something sufficiently URL-like at this point? Do you know @aherlihy @rose-m, or do you have any other ideas for how to do this well?

Reply:
As mentioned in Slack, the `uri` might contain multiple hosts in the seed list which is not URL compliant.
Furthermore, @mcasimir mentioned the escaping difference.
So I'm afraid we are stuck with re-structuring the URL again 😞 Maybe we can provide some better utility function and move it to a common dependency? Right now the `cli-repl` package knows quite well how to de-structure the URL and also uses `URL` to deal with parameter manipulation already.

Reply:
Also replaced this with a cleaner variant